x-pack/plugins/lens/public/visualizations/partition/toolbar.tsx

```diff
@@ -245,11 +248,9 @@ export function PieToolbar(props: VisualizationToolbarProps<PieVisualizationStat
         legendOptions={legendOptions}
         mode={layer.legendDisplay}
         onDisplayChange={onLegendDisplayChange}
-        valueInLegend={shouldShowValuesInLegend(layer, state.shape)}
-        renderValueInLegendSwitch={
-          'showValues' in PartitionChartsMeta[state.shape]?.legend ?? false
-        }
-        onValueInLegendChange={onValueInLegendChange}
+        legendStats={getLegendStats(layer, state.shape)}
+        allowLegendStats={'defaultLegendStats' in PartitionChartsMeta[state.shape]?.legend ?? false}
```

Comment:
See comment above. I'd propose to change here to:

```suggestion
        allowLegendStats={PartitionChartsMeta[state.shape].legend.defaultLegendStats ?? false}
```